web-ui/src/main/resources/catalog/components/common/map/mapsManager.js

```diff
@@ -182,10 +182,16 @@
             if (config.context) {
               mapReady = gnOwsContextService.loadContextFromUrl(
                 config.context, map);
+            } else {
+              //Failback needed for controllers that rely on map creation
+              mapReady = $q.defer();
+              mapReady.resolve();
```

Comment:
Very dummy promise, but there are other parts of the code waiting for a promise here

Reply:
You can simply use `$q.when(value)` to create a simple promise like that. See:
https://docs.angularjs.org/api/ng/service/$q#when